rust-escrow/dao-escrow/src/lib.rs

```diff
@@ -0,0 +1,185 @@
+use near_sdk::borsh::{self, BorshDeserialize, BorshSerialize};
+use near_sdk::collections::UnorderedMap;
+use near_sdk::{env, ext_contract, Gas, near_bindgen};
+use near_sdk::json_types::{Base64VecU8};
+use near_sdk::{AccountId, Balance, Promise, PromiseResult};
+
+/// Amount of gas for create action.
+pub const GAS_FOR_CREATE: Gas = Gas(90_000_000_000_000);
+pub const GAS_FOR_CREATE_CALLBACK: Gas = Gas(5_000_000_000_000);
+
+// define the methods we'll use on the other contract
+#[ext_contract(ext_dao_factory)]
+pub trait DaoFactory {
+    fn create(&mut self, name: String, args: Base64VecU8);
+}
+
+// define methods we'll use as callbacks on our contract
+#[ext_contract(ext_self)]
+pub trait MyContract {
+    fn on_create_callback(&mut self, country_code: String, daos_by_country: u128) -> bool;
+}
+
+#[near_bindgen]
+#[derive(BorshDeserialize, BorshSerialize)]
+pub struct DaoEscrow {
+    daos: UnorderedMap<String, u128>, // Country code and sequential number
+    dao_factory_account: AccountId
+}
+
+impl Default for DaoEscrow {
+    fn default() -> Self {
+        env::panic_str("DaoEscrow should be initialized before usage")
+    }
+}
+
+#[near_bindgen]
+impl DaoEscrow {
```

Comment:
I'm a bit hesitant about the name. `DaoEscrow` reads as if this works as an `Escrow` for a DAO, but it is actually a `DaoFactory`, or to be more specific `SputnikDao2Factory`. 

The difference is that this contract passes down the `deposit_accounts ` to create its members. 

Lets call the external contract trait: `ExtSputnikDao2Factory` and this one: `SputnikDao2Factory`. WDYT?

Reply:
Or Maybe only `daofactory`.

Reply:
Yeah, in this case: `ExtDaoFactory` and `DaoFactory`